pandas/core/window/rolling.py

```diff
@@ -93,6 +93,20 @@ def __init__(
         self.axis = obj._get_axis_number(axis) if axis is not None else None
         self.validate()
         self._numba_func_cache: Dict[Optional[str], Callable] = dict()
+        self.exclusions = kwargs.get("exclusions", set())
+
+    def _shallow_copy(self, obj: FrameOrSeries, **kwargs) -> ShallowMixin:
+        if isinstance(obj, ABCDataFrame):
+            try:
+                new_obj = super()._shallow_copy(
+                    obj, exclusions=self.exclusions.intersection(obj.columns), **kwargs
+                )
+                new_obj.obj = new_obj._obj_with_exclusions
+                return new_obj
+            except TypeError:  # Some _shallow_copy don't take `exclusions` as argument
```

Comment:
where? let's fix that instead ofworking around. try/excepts are terrible to do around things like this.